calculator_test.go

```diff
@@ -225,24 +227,28 @@ func TestSqrt(t *testing.T) {
 	}{
 		{
 			name:  "Calculate square root of a positive number",
-			want:  7,
 			input: 49,
 		},
+		{
+			name:  "Calculate square root of a long positive number",
+			input: 94339,
+		},
 		{
 			name:        "Calculate square root of a negative number",
 			errExpected: true,
-			want:        0,
 			input:       -49,
 		},
 	}
 	for _, tC := range testCases {
 		t.Run(tC.name, func(t *testing.T) {
 			got, err := calculator.Sqrt(tC.input)
-			if err != nil && tC.errExpected == false {
-				t.Fatalf("Cannot calculate square root of %f: %s", tC.input, err)
+			errReceived := err != nil
+			if tC.errExpected != errReceived {
+				t.Fatal(err)
```

Comment:
This is going to be a confusing failure output if the function _doesn't_ return an error when it's supposed to:

```
<nil>
```

Let's give the programmer some more help:

```suggestion
				t.Fatalf("Sqrt(%.2f): unexpected error status %v", tc.input, err)
```

This language covers both cases (when we didn't expect an error, unexpected error `something`, and when we did expect an error, unexpected error _status_ `nil`).

Reply:
By the way, it's always worth exercising your tests by making every possible failure case happen (either by tweaking the expectations, or by messing up the function itself), so that you can _see_ what it outputs and verify that it's accurate and helpful. Someday these tests will fail for real, and it will be unexpected, mysterious, and demotivating for the programmer involved. We want to do everything we can to help them—they might be us!

Reply:
right